apps/oc_chef_authz/src/oc_chef_policy.erl

```diff
@@ -0,0 +1,182 @@
+    list_policies_for_org.
+
+bulk_get_query() ->
+    %% TODO: do we need this?
+    ok.
+
+new_record(OrgId, AuthzId, PolicyData) ->
+    Name = ej:get({<<"name">>}, PolicyData),
+    Id = ej:get({<<"id">>}, PolicyData, chef_object_base:make_org_prefix_id(OrgId, Name)),
+    PolicyGroup = ej:get({<<"policy_group">>}, PolicyData),
+    #oc_chef_policy{
+       id = Id,
+       authz_id = AuthzId,
+                       org_id = OrgId,
+                       name = Name,
+                       policy_group = PolicyGroup,
+      serialized_object = PolicyData}.
+
+create_record(OrgId, Name, RequestingActorId) ->
+    Policy = #oc_chef_policy{
+                           org_id = OrgId,
+                           name = Name},
+    set_created(Policy, RequestingActorId).
+
+set_created(#oc_chef_policy{} = Object, ActorId) ->
+    Object#oc_chef_policy{last_updated_by = ActorId}.
+
+set_updated(#oc_chef_policy{} = Object, ActorId) ->
+    Object#oc_chef_policy{last_updated_by = ActorId}.
+
+is_indexed() ->
+    false.
+
+ejson_for_indexing(#oc_chef_policy{}, _EjsonTerm) ->
+   {[]}.
+
+update_from_ejson(#oc_chef_policy{} = Policy, PolicyData) ->
+    Name = ej:get({<<"name">>}, PolicyData, name(Policy)),
+    Group = ej:get({<<"policy_group">>}, PolicyData, policy_group(Policy)),
+    Policy#oc_chef_policy{name = Name,
+                         policy_group = Group}.
+
+fields_for_update(#oc_chef_policy{
+                     last_updated_by = LastUpdatedBy,
+                     serialized_object = SerializedObject
+                                 } = Policy) ->
+    [name(Policy), policy_group(Policy), LastUpdatedBy, SerializedObject].
+
+
+fields_for_fetch(#oc_chef_policy{org_id = OrgId} = Policy) ->
+    [name(Policy), policy_group(Policy), OrgId].
+
+record_fields() ->
+    record_info(fields, oc_chef_policy).
+
+list(#oc_chef_policy{org_id = OrgId}, CallbackFun) ->
+    CallbackFun({list_query(), [OrgId], rows}).
+
+update(#oc_chef_policy{
+                      org_id = _OrgId,
+                      authz_id = _PolicyAuthzId,
+                      last_updated_by = _AuthzId
+                     } = _Record, _CallbackFun) ->
+    1.
+
+
+parse_binary_json(Bin) ->
+    {ok, chef_json:decode_body(Bin)}.
+
+
+fetch(#oc_chef_policy{} = Record, CallbackFun) ->
+    chef_object:default_fetch(Record, CallbackFun).
+
+
+flatten(#oc_chef_policy{
+          id = Id,
+          authz_id = AuthzId,
+          org_id = OrgId,
+          name = Name,
+          policy_group = Group,
+          last_updated_by = LastUpdatedBy,
+          serialized_object = SerializedObject}) ->
+    Compressed = chef_db_compression:compress(oc_chef_policy, jiffy:encode(SerializedObject)),
+    [Id, AuthzId, OrgId, Name, Group, LastUpdatedBy, Compressed].
+
+
+assemble_policy_ejson(#oc_chef_policy{
+                         serialized_object = SerializedObject
+                        }, _OrgName) ->
+    chef_db_compression:decompress(SerializedObject).
```

Comment:
Name should already be in the json and group should be excluded as it is not part of the policyfile itself.